sandy/gls.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+"""
+This module contains all classes and functions specific for processing GLS.
+"""
+import numpy as np
+import pandas as pd
+import sandy
+
+__author__ = "Aitor Bengoechea"
+__all__ = [
+        "GLS",
+        ]
+
+
+class GLS():
+    """
+    Container for processing GLS.
+    """
+
+    def __repr__(self):
+        return self.data.__repr__()
+
+    def __init__(self, *args, **kwargs):
+        self.data = pd.Series(*args, **kwargs)
+
+    @property
+    def data(self):
+        return self._data
+
+    @data.setter
+    def data(self, data):
+        self._data = pd.Series(data, dtype=float)
+
+    @classmethod
+    def from_vec(cls, vec):
```

Comment:
You don't need this...initializing with `from_vec` or with `__init__` do the same